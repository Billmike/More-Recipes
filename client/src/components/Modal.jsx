import React from 'react';
import Modal from 'react-modal';

const DeleteModal = props => (
  <Modal
    isOpen={!!props.selectedRecipe}
    contentLabel="Delete Recipe?"
    onRequestClose={props.handleClearSelectedRecipe}
    closeTimeoutMS={300}
    className="modal-class"
    ariaHideApp={false}
  >
    <h3 className="modal-ptag"> Delete Recipe </h3>
    <p className="modal-ptag"> Are you sure want to delete this recipe?</p>
    <div className="group-btn">
      <button
        className="btn btn-danger user-btn modal-btn-red"
        onClick={props.onRemoveRecipe}
      >
        Yes
      </button>
      <button
        className="btn user-btn modal-btn-white"
        onClick={props.handleClearSelectedRecipe}
      >
        No
      </button>
    </div>
  </Modal>
);

export default DeleteModal;
