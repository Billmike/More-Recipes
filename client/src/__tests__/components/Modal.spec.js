import React from 'react';
import { shallow } from 'enzyme';
import Modal from '../../components/Modal';

describe('<Modal />', () => {
  it('Should render the Modal correctly', () => {
    const wrapper = shallow(<Modal />);
    expect(wrapper).toMatchSnapshot();
  });
});
