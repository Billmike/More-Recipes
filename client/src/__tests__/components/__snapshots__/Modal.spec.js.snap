// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Should render the Modal correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal-class"
  closeTimeoutMS={300}
  contentLabel="Delete Recipe?"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal-ptag"
  >
     Delete Recipe 
  </h3>
  <p
    className="modal-ptag"
  >
     Are you sure want to delete this recipe?
  </p>
  <div
    className="group-btn"
  >
    <button
      className="btn btn-danger user-btn modal-btn-red"
    >
      Yes
    </button>
    <button
      className="btn user-btn modal-btn-white"
    >
      No
    </button>
  </div>
</Modal>
`;
